bundles/org.openhab.voice.whisperstt/README.md

```diff
@@ -0,0 +1,246 @@
+
+The following platforms are supported:
+
+* Windows10 x86_64
+* Debian GLIBC x86_64/arm64 (min GLIBC version 2.31 / min Debian version Focal)
+* macOS x86_64/arm64 (min version v11.0)
+
+The native binaries for those platforms are included in this add-on provided with the openHAB distribution.
+
+## CPU compatibility
+
+To use this binding it's recommended to use a device at least as powerful as the RaspberryPI 5 with a modern CPU.
+The execution times on Raspberry PI 4 are x2, so just the tiny model can be run on under 5 seconds.
+
+If you are going to use the binding in a `x86_64` host the CPU should support the flags: `avx2`, `fma`, `f16c`, `avx`.
+You can check those flags on linux using the terminal with `lscpu`.
+You can check those flags on Windows using a program like `CPU-Z`.
+
+If you are going to use the binding in a `arm64` host the CPU should support the flags: `fphp`.
+You can check those flags on linux using the terminal with `lscpu`.
+
+## Transcription time
+
+On a Raspberry PI 5, the approximate transcription times are:
+
+| model      | exec time |
+| ---------- | --------: |
+| tiny.bin   |      1.5s |
+| base.bin   |        3s |
+| small.bin  |      8.5s |
+| medium.bin |       17s |
+
+
+## Configuring the model
+
+Before you can use this service you should configure your model.
+
+You can download them from the sources provided by the [whisper.cpp](https://github.com/ggerganov/whisper.cpp) author:
+
+* https://huggingface.co/ggerganov/whisper.cpp
+* https://ggml.ggerganov.com
+
+You should place the downloaded .bin model in '\<openHAB userdata\>/whisper/' so the add-ons can find them.
+
+Remember to check that you have enough RAM to load the model, estimated RAM consumption can be checked on the huggingface link.
+
+## Using alternative whisper.cpp library
+
+It's possible to use your own build of the whisper.cpp shared library with this add-on.
+
+On `Linux/macOs` you need to place the `libwhisper.so/libwhisper.dydib` at `/usr/local/lib/`.
+
+On `Windows` the `whisper.dll` file needs to be placed in any directory listed at the variable `$env:PATH`, for example `X:\\Windows\System32\`.
+
+In the [Whisper.cpp](https://github.com/ggerganov/whisper.cpp) README you can find information about the required flags to enable different acceleration methods on the cmake build and other relevant information.
+
+Note: You need to restart openHAB to reload the library.
+
+## Grammar
+
+The whisper.cpp library allows to define a grammar to alter the transcription results without fine-tuning the model.
+
+Internally whisper works by inferring a matrix of possible tokens from the audio a then resolving the final transcription from it using the Greedy or Bean Search algorithms,
+the grammar feature allows you to modify the probabilities of the tokens inferred adding a penalty to the tokens outside the grammar so the transcription gets resolved on a different way.
```

Comment:
**The** grammar...
... of the **inferred tokens by** adding a penalty
... so **that** the transcription ...
... **in** a different way.